internal/venice-common/src/main/java/com/linkedin/venice/serialization/avro/InternalAvroSpecificSerializer.java

```diff
@@ -50,7 +52,14 @@ private static class ReusableObjects {
   private static final ThreadLocal<ReusableObjects> threadLocalReusableObjects =
       ThreadLocal.withInitial(ReusableObjects::new);
 
-  /** Used to configure the {@link #schemaReader}. */
+  /**
+   * Used to configure the {@link #schemaReader}.
+   *
+   * Deprecated: This path has now been superseded by {@link #setSchemaReader(SchemaReader)}, which is used everywhere
+   *             except in {@link KafkaConsumerFactoryImpl#getRecordKafkaConsumer()}. Once that usage is also eliminated
+   *             we could remove the config from here.
+   */
+  @Deprecated
   public static final String VENICE_SCHEMA_READER_CONFIG = "venice.schema-reader";
```

Comment:
Can we completely get rid of this config?
For example, incremental ETL could consume the data at any point, and it is not guaranteed that incremental ETL will get the schema header in the Control Message, and even it gets it, it may not have a way to persist it locally for use in next restart.

Reply:
I'd like to get rid of it but IDK if doing it now is ideal. It requires some more refactoring in `PartitionOffsetFetcherImpl` in order to achieve it, and I think @haoxu07 is going to change the code structure a bit more in the next phase of this work, so it might be simpler to wait until then, otherwise we may need to refactor this multiple times.

Reply:
I see the comment now.
My initial comment is that we shouldn't remove `SchemaReader` completely from this class and since there is a `setter`, we are fine.

Reply:
Right, we are aligned. The SchemaReader and the schema coming from headers are two alternatives for achieving forward compat and they are complementary, not mutually-exclusive.

Reply:
Right, we are aligned. The SchemaReader and the schema coming from headers are two alternatives for achieving forward compat and they are complementary, not mutually-exclusive.